### ISSUE TITLE
Two tests failing in latest master HEAD ca76e7c

### ISSUE THREAD

**Issue by dwvisser** [dependency-check/DependencyCheck#273]
These are the failures I'm seeing at commit ca76e7c. I am seeing a similar "No documents exist" error when I build using `-DskipTests`, and invoke the command-line application.

```
Tests run: 4, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.362 sec <<< FAILURE! - in org.owasp.dependencycheck.data.update.BaseUpdaterTest
testGetProperties(org.owasp.dependencycheck.data.update.BaseUpdaterTest)  Time elapsed: 0.483 sec  <<< FAILURE!
java.lang.AssertionError: null
    at org.owasp.dependencycheck.data.update.BaseUpdaterTest.testGetProperties(BaseUpdaterTest.java:55)

Running org.owasp.dependencycheck.analyzer.HintAnalyzerTest
[INFO] Central analyzer disabled
[ERROR] No documents exist

Unable to continue dependency-check analysis.
Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.602 sec <<< FAILURE! - in org.owasp.dependencycheck.analyzer.HintAnalyzerTest
testAnalyze(org.owasp.dependencycheck.analyzer.HintAnalyzerTest)  Time elapsed: 0.602 sec  <<< FAILURE!
java.lang.AssertionError: null
    at org.owasp.dependencycheck.analyzer.HintAnalyzerTest.testAnalyze(HintAnalyzerTest.java:109)
```

**Comment by jeremylong** [comment]
Lovely... built with no errors on my system and there were no errors on the Cloudbees build: https://dependency-check.ci.cloudbees.com/job/dependency-check/276/

I'll have time to look at this more in the morning... `mvn -X clean install` and email me the log?

**Comment by colezlaw** [comment]
Things work for me under Oracle JDK 1.7 and 1.8 on Mac.

It doesn't LOOK like those tests attempt to download anything, but I
believe I've seen those issues when behind a proxy and it's unable to load
CVE data from NVD. I'm guessing you're not behind a proxy, though?

**Comment by dwvisser** [comment]
The machine having the issue is running Ubuntu 15.04/OpenJDK 1.8. I'll post build logs later.

**Comment by dwvisser** [comment]
I am behind a an HTTPS proxy, which sometimes fatally slows down CVE downloads, but not usually. Also, I think I was running with the --noupdate option.

**Comment by dwvisser** [comment]
I have successfully built and run using `mvn -X clean install` on another machine:
- Ubuntu 14.04
- Oracle JDK 1.8
- No proxy that I know of
- at latest master commit 74ab678.

That same machine definitely showed the failure initially with just a `mvn install`. I also tried `mvn clean install` but I think I aborted it before it could show the problem. I'm _half_ certain that I already tried a `mvn clean install` on the original box. I will go back to that and try again.

**Comment by dwvisser** [comment]
I temporarily took my machine off my work network, so no more proxy AFAIK. However, on this machine (Ubuntu 15.04, OpenJDK 1.8) this problem is persisting. The output from `mvn -X clean install` is here: https://gist.github.com/dwvisser/d7553c5e9077699418f5

**Comment by hansjoachim** [comment]
Can confirm. I get failures in these tests as well:

```
Results :

Failed tests: 
  HintAnalyzerTest.testAnalyze:109 null
  BaseUpdaterTest.testGetProperties:55 null

Tests run: 200, Failures: 2, Errors: 0, Skipped: 9
```

Latest master (d55392ed7121f6f1656745ed2ad0ea5306383abb) / Ubuntu 15.04 / Openjdk-8 8u45-b14-1 / no proxy.

However... the test names sound familiar. I ran into similar problems with HintAnalyzerTest, but I'm fairly sure those were fixed at some point. I think I first mentioned it towards the end of #177, but I think a follow-up issue was filed (possibly #181?).

Regarding the cloudbees job, that seems to build with java7, so I don't think that would have caught this issue. If possible, it might be useful to set up an additional job so that latest master can be built with jdk7 and jdk8.

**Comment by dwvisser** [comment]
I cloned to another one of my VMs:
- Fedora 22
- OpenJDK 8
- Still off work network, so proxies likely not a cause

I have trouble getting `mvn clean install` to complete, but that seems to be due to connection issues with the vulnerabilities server. It appeared that at least one of the two tests had already passed. I built with `mvn clean install -DskipTests`. Runing the CLI app worked, except that vulnerability downloading got bogged down. I did a CTRL-C, and restarted with `--noupdate`, and everything ran fine, i.e., no "No documents exist" error.

Given the reproduction by @hansjoachim, it appears that the magic failure combination is OpenJDK 8 plus Ubuntu 15.04. I will see if installing Oracle JDK 1.8 fixes the issue.

**Comment by hansjoachim** [comment]
@dwvisser Interesting. Could you check which version of OpenJDK 8 was used Fedora?

Ubuntu 15.04 has 8u45 in the default archives, but the current development release of Ubuntu (codenamed Wily) recently added 8u60. I'm trying to check with that in case it's a problem in some minor version, but currently I'm struggling with maven and lucene for some reason.

**Comment by dwvisser** [comment]
On my Fedora VM: `uname -r` yields "4.0.6-300.fc22.x86_64", and `java -version` yields "1.8.0_45".

**Comment by dwvisser** [comment]
Well, now I've verified that using Oracle JDK 8 on my Ubuntu 15.04 box did _not_ solve the issue.

**Comment by hansjoachim** [comment]
FWIW, I get the same failures with OpenJDK 8u60 on Ubuntu Wily. In fact, I get a couple of errors as well. On a slightly more positive note, it looks like at least HintAnalyzerTest failed on the CI system too https://dependency-check.ci.cloudbees.com/job/dependency-check/281/.

**Comment by dwvisser** [comment]
Alas, it looks like a different failure. I have now also tried OpenJDK 7 on the Ubuntu system, and get the same two test failures that I did with OpenJDK/Oracle Java 8. Since I need to move on from this, I will be moving my dev environment for this project over to the Fedora VM.

**Comment by dwvisser** [comment]
This all may be a network issue on NIST's end, coupled to some inefficiency introduced in the recent commits. I reverted my Ubuntu 15.04 box to OpenJDK 8, did a `mvn clean install -DskipTests`, and ran the command-line. I got the "No documents exist" error. I had run with `--noupdate`, so I removed that option and tried again. The entire NVD CVE database was sucked down (much faster than I had been seeing this morning), and the analysis completed successfully.

However `mvn clean install` still results in the failures at HintAnalyzerTest.testAnalyze:109 and BaseUpdaterTest.testGetProperties:55.

I vote +1 for #268. Once that fix is in, I will make sure to have a mirror set up so I can rule out network issues in the future.

**Comment by jeremylong** [comment]
The HintAnalyzer does not call out to NIST - so while there might be issues with the proxy I doubt it is causing the build failures... What is better is that the build failures on the CI are resolved. However, I'm still not sure what was causing the issue - I added a null check, but that code should not have been running AND I added logging statements around the code that appeared to be failing and they did not fire...  I love mysterious intermittent bugs.

So - with the latest are you still seeing the HintAnalyzer fail?

**Comment by hansjoachim** [comment]
I still see the nullpointers in HintAnalyzerTest and BaseUpdaterTest in cf4771ca0037507c144ae09f593aa6e90d55c9ec with OpenJKD8 / Ubuntu.

**Comment by dwvisser** [comment]
+1 same cf4771c, same errors still showing on Ubuntu 15.04/OpenJDK 8, but not on Fedora 22, also running OpenJDK 8.

**Comment by javabeanz** [comment]
Using the Travis automated build, you can select several jdks for build plan. Does Clowdbees have this too ?

**Comment by dwvisser** [comment]
I don't know if Cloudbees allows this. In my own testing (see above), this affects Ubuntu 15.04+, regardless of choice of JDK. Judging from command-line logs, it looks like Cloudbees builds are run in a Fedora-flavored container.

**Comment by dwvisser** [comment]
Errors persist at latest commit 48b5e79 on Ubuntu 15.04, OpenJDK 1.8.0:

```
$ java -version
openjdk version "1.8.0_45-internal"
OpenJDK Runtime Environment (build 1.8.0_45-internal-b14)
OpenJDK 64-Bit Server VM (build 25.45-b02, mixed mode)
```

**Comment by jeremylong** [comment]
I am still setting up an Ubuntu box - however, I did find a bug in the code
that was causing failures on our CI. Can anyone confirm if the latest code
base is still failing to build on Ubuntu w/ OpenJDK 1.8? (note, you may see
expected errors in the console during the build - the logging in test was
cranked up to debug this and hasn't been turned back down yet).

--Jeremy

**Comment by dwvisser** [comment]
No luck. Still seeing exact same issue. My environment (fully patched Ubuntu MATE 15.04, FWIW):

```
$ uname -a; java -version; git log -1
Linux dale-VirtualBox 3.19.0-22-generic #22-Ubuntu SMP Tue Jun 16 17:15:15 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
openjdk version "1.8.0_45-internal"
OpenJDK Runtime Environment (build 1.8.0_45-internal-b14)
OpenJDK 64-Bit Server VM (build 25.45-b02, mixed mode)
commit 59512e236bbad06cd043291c34036e2f9230d19c
Author: Jeremy Long <jeremy.long@gmail.com>
Date:   Sun Jul 19 06:21:30 2015 -0400

    set objects to null after closing them
```

**Comment by jeremylong** [comment]
Please confirm that this issue has been resolved. The code compiles and tests without errors on my local ubuntu w/ jdk 8.

**Comment by hansjoachim** [comment]
Built successfully on Ubuntu Wily with openJDK 8u60. :)

Could you elaborate a bit on what the issue/fix was?

**Comment by jeremylong** [comment]
I had added a test case that utilized the H2 database - but I forgot to extend the BaseDBTestCase (I only extended the BaseTestCase). As such, a new empty database was created instead of copying an existing one from the test resources. Because a new empty database was there other test cases just assume the database is okay (i.e. they don't copy over an existing db) the other subsequent tests then failed because there was no data.  

Why this showed up in Ubuntu with OpenJDK8 vs. Windows 7 JDK 8 - no idea. I can only assume test order. The new test ran later then other tests that use the database.

--Jeremy
